src/v2/controllers/RegistrationFormController.js

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) 2020, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import FormController from './FormController';
+import { FORMS } from '../ion/RemediationConstants';
+
+export default FormController.extend({
+  postRender() {
+    // TODO: what if user navigate to this page but there is valid state token
+    // in session storage
+    // - try to sign-in
+    // - during at middle of sign-in, change URL to /signin/register/
+    // user probably expects to see sign-up page
+    // actually widget will continue show sign-in flow (because of the saved state token)
+    if (this.options.appState.get('currentFormName') === FORMS.IDENTIFY) {
+      // 1. auto switch to porfile enroll (register) page
+      // basically mimic the flow that show identify page and click sign-up link.
+      if (this.options.appState.hasRemediationObject(FORMS.SELECT_ENROLL_PROFILE)) {
+        this.handleInvokeAction(FORMS.SELECT_ENROLL_PROFILE);
+      } else {
+        const messages = {
+          'type': 'array',
+          value: [
+            {
+              message: 'registration is not supported',
```

Comment:
nit: this looks like a custom client side error ? let us make sure to add i18n key to keep it consistent with the rest of our API responses.

Reply:
Surely yes!